handler/handler.go

```diff
@@ -18,6 +24,43 @@ type NatsQueue struct {
 	Topic     string
 }
 
+func (q *NatsQueue) connect() error {
+	nc, err := stan.Connect(
+		q.ClusterID,
+		q.ClientID,
+		stan.NatsURL(q.NATSURL),
+		stan.SetConnectionLostHandler(func(conn stan.Conn, err error) {
+			log.Printf("Disconnected from %s\n", q.NATSURL)
+
+			q.reconnect(0)
+		}),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	q.nc = nc
```

Comment:
This could cause a race between this and the `NatsQueue.Queue()` call. To avoid the data race I would add a RWMutex to NatsQueue and use the write lock here and an RLock/RUnlock in Queue() to capture `q.nc` and publish outside of the lock:
```
q.RLock()
nc := q.nc
q.RUnlock()
return q.nc.Publish(q.Topic, out)
```

Reply:
Good idea!

Reply:
@kozlovic I have implemented that. Can you check now?

Reply:
Thanks for the feedback on this @kozlovic